template/scripts/buildProd.ts

```diff
@@ -0,0 +1,44 @@
+export {};
+const spawn = require('child_process').spawn;
+
+const bisonConfig = require('../package.json').bison;
+const DEFAULT_BUILD_COMMAND = `yarn build:nexus && yarn build:prisma && yarn build:next`;
+
+/**
+ * This builds the production app.
+ * if the current branch should be migrated, it runs migrations first.
+ */
+function buildProd() {
+  const { staging, production } = bisonConfig.branches;
+  const branchesToMigrate = new RegExp(`${staging}|${production}`);
+  const currentBranch = process.env.BRANCH || process.env.VERCEL_GITHUB_COMMIT_REF;
+  const shouldMigrate = branchesToMigrate.test(currentBranch);
+
+  console.log('--------------------------------------------------------------');
+  console.log('Determining if we should migrate the database...');
+  console.log('branches to migrate:', branchesToMigrate);
+  console.log('current branch name:', currentBranch || '(unknown)');
+  console.log(shouldMigrate ? `${currentBranch} detected. Migrating.` : `Not running migrations.`);
+  console.log('--------------------------------------------------------------');
```

Comment:
Users will see this log in the build, so it should be really clear. This isn't supported on Heroku, so we'll have to add the way to do it for that platform at some point.

Reply:
👍